karapace/schema_registry_apis.py

```diff
@@ -395,16 +420,39 @@ async def subjects_list(self, content_type):
         subjects_list = [key for key, val in self.ksr.subjects.items() if self.ksr.get_schemas(key)]
         self.r(subjects_list, content_type, status=HTTPStatus.OK)
 
-    async def subject_delete(self, content_type, *, subject):
-        self._subject_get(subject, content_type)
-        version_list = list(self.ksr.get_schemas(subject))
+    async def subject_delete(self, content_type, *, subject, request: HTTPRequest):
+        permanent = request.query.get("permanent", "false").lower() == "true"
+
+        subject_data = self._subject_get(subject, content_type, include_deleted=permanent)
+
+        if permanent is True and [
```

Comment:
`if permanent  and`

No need for `is True`.

Reply:
please move this to a validation method, lets try to end up with something like this:

https://github.com/aiven/karapace/blob/ac9a811c9af08afc86237d18d894a071edd25584/karapace/schema_registry_apis.py#L618-L631

Otherwise it gets convoluted and hard to read.

Reply:
Validation method for delete case is quite different from others as it needs to read soft deleted versions and `permanent` flag so I failed to see how mixing this into another method really improves readability of the code...

Reply:
I'm not proposing mixing code. It is not about code reuse, but about code organization, making sure the methods are short and giving names to code so that it is easier to follow. i.e. creating a method `_validate_existing_schemas_are_soft_deleted` with this if condition. That makes the method shorter, and the function name is good enough documentation for what is the intent of that piece of code.